### PR TITLE
Verify & create user profile

### PR DESCRIPTION
**Pull request by Jude-adesulu**
The work flow
- A verification link(containing a token) will be sent to user at the time of acc creation
- Once the mail link is clicked by the user, an endpoint is waiting to consume the token and verify
- Upon verification, a call to axios request for a profile to be created to user
- The profile is then returned to the user
- Also verification is resend to a user on account recreating if mail wasn't confirmed the first time

WARNING!!! I'm yet to generate a migration file, do well and generate before use

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -27,3 +27,10 @@ TYPEORM_MIGRATIONS = src/database/migration/**/*.ts
 TYPEORM_MIGRATIONS_DIR = src/database/migration
 TYPEORM_SUBSCRIBERS = src/database/subscriber/**/*.ts
 TYPEORM_SUBSCRIBERS_DIR = src/database/subscriber
+
+
+IDENTITY_SERVICE_BASE_URL=""
+CORE_SERVICE_BASE_URL=""
+NOTIFICATION_URL=NOTIFICATION_BASE_URL/email/send-mail
```
Comment (PhilemonBrain) [inline .env.example:34]: the NOTIFICATION_URL should be the base url only. the extra part which is the route should be in ur code please

File: src/common/constants/config.constant.ts [hunk 2]
```diff
@@ -8,9 +8,13 @@ export const configConstant = {
...
     timeout: 'HTTP_TIMEOUT',
-    maxRedirects: 'HTTP_MAX_REDIRECTS'
+    maxRedirects: 'HTTP_MAX_REDIRECTS',
   },
   profileUrl: {
-    baseUrl: 'CORE_SERVICE_BASE_URL'
-  }
+    baseUrl: 'CORE_SERVICE_BASE_URL',
+  },
+  baseUrl: {
```
Comment (PhilemonBrain) [inline src/common/constants/config.constant.ts:16]: this might be better as baseUrl.notification not baseUrl.notificationUrl

**Review by PhilemonBrain — CHANGES_REQUESTED** [review]

File: src/email-verification/email-verification.controller.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+@Controller('email-verification')
+export class EmailVerificationController {
+  constructor(
+    private readonly emailVerificatioService: EmailVerificationService,
+  ) {}
+
+  @Get('/:token')
+  async verifyEmail(@Param() emailTokenDto: EmailTokenDto) {
+    return await this.emailVerificatioService.decodeEmailToken(
+      emailTokenDto.token,
```
Comment (Chimexx) [inline src/email-verification/email-verification.controller.ts:14]: Pass this into a custom return object, check other controller files

File: src/email-verification/email-verification.service.ts [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+    private readonly httpService: HttpService,
+  ) {}
+
+  /*
+   * sendVerificationLink - send verification token to a newly registered user
+   * @Params: email - email created by the user
+   * return - return a http request to send email notification
+   */
+  async sendVerificationLink(email: string) {
+    const payload: VerifyToken = { email };
```
Comment (Chimexx) [inline src/email-verification/email-verification.service.ts:29]: Is payload neccessary? just use email

File: .env.example (hunk 1 above)
Comment (Jude-adesulu) [reply .env.example:34]: Okay noted.  I'm on it

**Review by Chimexx — CHANGES_REQUESTED** [review]

File: src/email-verification/email-verification.service.ts (hunk 4 above)
Comment (Jude-adesulu) [reply src/email-verification/email-verification.service.ts:29]: I was having some errors when I passed it directly, that's why I used this method.  I will see if I can get it changed

File: src/email-verification/email-verification.controller.ts (hunk 3 above)
Comment (Jude-adesulu) [reply src/email-verification/email-verification.controller.ts:14]: > 

I just checked auth controller I didn't see any customer respone for it

Comment (PhilemonBrain) [reply src/email-verification/email-verification.controller.ts:14]: @Jude-adesulu if you check the dummy-core service, you'll see that we wrap the resposnse from the service methods in a ZAResponse.Ok object like this : 

const example = await this.emailVerificatioService.decodeEmailToken(
      emailTokenDto.token)
return ZapiResponse.Ok(example, 'Ok', '200')

File: src/email-verification/email-verification.service.ts (hunk 4 above)
Comment (PhilemonBrain) [reply src/email-verification/email-verification.service.ts:29]: @Jude-adesulu its okay this way if it works.

**Review by PhilemonBrain — CHANGES_REQUESTED** [review]

**Review by PhilemonBrain — APPROVED** [review]

**Pull request merged**